src/providers/krb5/krb5_utils.c

```diff
@@ -570,6 +573,30 @@ parse_krb5_map_user(TALLOC_CTX *mem_ctx, const char *krb5_map_user,
         }
     }
 
+    /* conversion names to fully-qualified names */
+    for (int i = 0; i < size; i++) {
+        fq_name = sss_create_internal_fqname(tmp_ctx,
+                                             name_to_primary[i].id_name,
+                                             dom_name);
+        if (fq_name == NULL) {
+            DEBUG(SSSDBG_OP_FAILURE,
+                  "sss_create_internal_fqname failed\n");
+            ret = ENOENT;
+            goto done;
+        }
+        name_to_primary[i].id_name = talloc_steal(name_to_primary, fq_name);
```

Comment:
I think this is just not needed, why not assign to the id_name/krb5_primary and remove the fq_name variable?

Something like:
name_to_primary[i].krb_primary = sss_create_internal_fqname(name_to_primary,